components/datetime/src/pattern/mod.rs

```diff
@@ -107,8 +122,161 @@ impl From<Vec<PatternItem>> for Pattern {
     }
 }
 
+impl fmt::Display for Pattern {
```

Comment:
I'm a bit confused by this feedback, as the `Writeable` trait needs to know the size of the output ahead of time. In the example of `WelcomeMessage`, this is easy to determine as it's `8 + self.name.len()`. However, here this is deriving a string. In order to know the length, the string must be generated first.

It looks like the closest example in the code may be the `FixedDecimal` `Writeable`. This is duplicating the branching logic but determining the string size rather than the string itself. Another similar example is the `FormattedDateTime`. However, there is a TODO as I'm guessing this will be quite complex as well.

To me this does not seem like a good use case for `Writeable`, as the determination of string size is not a particularly trivial operation, and this is not hot code.

I was somewhat hesitant to convert this over to `fmt::Display`, as the need for this code was to convert it to a string, while the `fmt::Display` is not particularly in the current requirements.

The action I'm going to take here is to add a note about Writeable for future readers with my analysis.

Reply:
My analysis suggests that computing the length of the string can fairly substantially enhance performance, and it's not usually as hard to do as you think.  But, this particular example is low priority and not a good use of time.  So just leave the comment and it is fine.